test/algorithms/test_partial_equivalence.cpp

```diff
@@ -0,0 +1,420 @@
+  // 8 qubits, 3 data qubits
+  qc::QuantumComputation c2{"./circuits/period_finding_2.qasm"};
+
+  // 3 measured qubits and 3 data qubits
+
+  c2.setLogicalQubitAncillary(7);
+  c2.setLogicalQubitGarbage(7);
+  c2.setLogicalQubitAncillary(6);
+  c2.setLogicalQubitGarbage(6);
+  c2.setLogicalQubitAncillary(5);
+  c2.setLogicalQubitGarbage(5);
+  c2.setLogicalQubitAncillary(3);
+  c2.setLogicalQubitGarbage(3);
+  c2.setLogicalQubitAncillary(4);
+  c2.setLogicalQubitGarbage(4);
+
+  c1.setLogicalQubitAncillary(7);
+  c1.setLogicalQubitGarbage(7);
+  c1.setLogicalQubitAncillary(6);
+  c1.setLogicalQubitGarbage(6);
+  c1.setLogicalQubitAncillary(5);
+  c1.setLogicalQubitGarbage(5);
+  c1.setLogicalQubitAncillary(3);
+  c1.setLogicalQubitGarbage(3);
+  c1.setLogicalQubitAncillary(4);
+  c1.setLogicalQubitGarbage(4);
+  EXPECT_TRUE(dd::partialEquivalenceCheck(c1, c2, dd));
+}
+
+void partialEquivalencCheckingBenchmarks(
+    const std::unique_ptr<dd::Package<dd::DDPackageConfig>>& dd,
+    const size_t minN, const size_t maxN, const size_t reps,
+    const bool addAncilla) {
+  for (size_t n = minN; n < maxN; n++) {
+    std::chrono::microseconds totalTime{0};
+    std::uint16_t totalGates{0};
+    for (size_t k = 0; k < reps; k++) {
+      dd::Qubit d{0};
+      if (addAncilla) {
+        d = static_cast<dd::Qubit>(rand()) % static_cast<dd::Qubit>(n - 1) + 1;
+      } else {
+        d = static_cast<dd::Qubit>(n);
+      }
+      dd::Qubit m{0};
+      if (d == 1) {
+        m = 1;
+      } else {
+        m = static_cast<dd::Qubit>(rand()) % static_cast<dd::Qubit>(d - 1) + 1;
+      }
+      const auto [c1, c2] = dd::generateRandomBenchmark(n, d, m);
+
+      const auto start = std::chrono::high_resolution_clock::now();
+      const bool result = dd::partialEquivalenceCheck(c1, c2, dd);
+      // Get ending timepoint
+      const auto stop = std::chrono::high_resolution_clock::now();
+      const auto duration =
+          std::chrono::duration_cast<std::chrono::microseconds>(stop - start);
+
+      EXPECT_TRUE(result);
+
+      // std::cout << "\nnumber of qubits = " << n << "; data qubits = " << d
+      //           << "; measured qubits = " << m
+      //           << "; number of gates = " << c2.size() << "\n";
+      // std::cout << "time: " << static_cast<double>(duration.count()) /
+      // 1000000.
+      //           << " seconds\n";
+      totalTime += duration;
+      totalGates += static_cast<std::uint16_t>(c2.size());
+    }
+    std::cout << "\nnumber of qubits = " << n << "; number of reps = " << reps
+              << "; average time = "
+              << (static_cast<double>(totalTime.count()) /
+                  static_cast<double>(reps) / 1000000.)
+              << " seconds; average number of gates = "
+              << (static_cast<double>(totalGates) / static_cast<double>(reps))
+              << "\n";
+  }
+}
+
+TEST(PartialEquivalenceTest, Benchmark) {
+  const auto dd = std::make_unique<dd::Package<>>(20);
+  srand(55);
```

Comment:
I am really surprised that clang-tidy is not complaining throughout this file about the direct use of `rand` and friends.
I'd much rather prefer to use the modern C++ equivalents of these constructs form `<random>`. Most of the things here can be solved with `uniform_int_distribution`.
Note that every `QuantumComputation` that is initiated has a properly seeded RNG generator. You can just use `qc.getGenerator()` to get it.

This comment naturally also applies to the code where these random benchmarks are generated.